File: ruby/fibonacci.rb
# frozen_string_literal: true

def fibFast(num, memo = {})
  return memo[num] unless memo[num].nil?

  return 1 if num <= 2

  memo[num] = fibFast(num - 1, memo) + fibFast(num - 2, memo)
end

puts "FIB: #{fastFibonacci(50)}"

File: go/fibonacci.go
package main

mport "fmt"

func fibonacci(num int, memo map[int]int) int {
	if num <= 1 {
		return 1
	}

	if val, ok := memo[num]; ok {
		return val
	}

	memo[num] = fibonacci(num-1, memo) + fibonacci(num-2, memo)
	return memo[num]
}

func main() {
	memo := make(map[int]int)
	fmt.Printf("FIB: %d\n", fibonacci(50, memo))
}
